### PR TITLE
feat: support struct expression

### PR DESCRIPTION
**Pull request by cykbls01**
## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

This PR implements the expression (function call) constructing a Row.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
https://github.com/singularity-data/risingwave/issues/2721

### CODE REVIEW

File: src/expr/src/expr/expr_combine.rs [hunk 1]
```diff
@@ -28,39 +29,50 @@ use risingwave_pb::expr::ExprNode;
...
+            )?;
+            chunk
+                .rows()
+                .try_for_each(|row| builder.append_row_ref(row))?;
+            builder.finish().map(|a| Arc::new(ArrayImpl::List(a)))
+        } else {
+            let mut builder = self.data_type.create_array_builder(input.capacity())?;
+            columns.iter().try_for_each(|c| builder.append_array(c))?;
+            builder.finish().map(Arc::new)
+        }
```
[the remark this answers is not in the document]
Comment (cykbls01) [reply src/expr/src/expr/expr_combine.rs:75]: Maybe Unnest(Array[1,2,3,4])?

File: e2e_test/batch/types/struct_ty.slt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+----
+(1,5,21)
+
+query T
+select Row(null);
+----
+(NULL)
+
+statement error
+select Row();
```
Comment (neverchanje) [inline e2e_test/batch/types/struct_ty.slt:20]:
```
postgres=# select row();
 row
-----
 ()
(1 row)
```

pg supports this sql.

Comment (neverchanje) [inline e2e_test/batch/types/struct_ty.slt:20]:
```
postgres=# select row();
 row
-----
 ()
(1 row)
```

pg supports this sql.

File: proto/expr.proto [hunk 3]
```diff
@@ -84,6 +84,7 @@ message ExprNode {
     // Array expression.
     ARRAY = 521;
     ARRAY_ACCESS = 522;
+    STRUCT = 523;
```
Comment (neverchanje) [inline proto/expr.proto:87]: `ROW` is better since we use `row` in syntax.

**Review by neverchanje — APPROVED** [review]
rest lgtm

**Pull request merged**